src/query/block/types.go

```diff
@@ -22,13 +22,17 @@ package block
 
 import (
 	"fmt"
+	"math"
 	"time"
 
 	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/ts"
 )
 
 // Block represents a group of series across a time bound
 type Block interface {
+	// Unconsolidated returns the unconsolidated version of the block
+	Unconsolidated() (UnconsolidatedBlock, error)
```

Comment:
Even for things which can support unconsolidated, you can still get an error. So we need an error here neverthless. I like this over having a bool so that clients don't check 2 things